### PR TITLE
Catalog grid section - #163

### PR DESCRIPTION
**Pull request by ahmed-belal**
#### What are the relevant tickets?
[Ticket](https://trello.com/c/PsnxTAjM/163-home-page-bootcamps-grid-catalog)

#### What's this PR do?
Adds a `CatalogGridPage` section to `HomePage`.

#### How should this be manually tested?
Add the `CatalogGridPage` to `HomePage` and add content to it. Verify the home page displays the content as expected.

#### Screenshots (if appropriate)
Wagtail editing interface
Screenshot 2020-06-16 at 2 52 00 PM

Mobile layout
screencapture-bc-odl-local-8099-2020-06-16-14_53_19

Tablet layout
screencapture-bc-odl-local-8099-2020-06-16-14_54_04

Desktop layout
screencapture-bc-odl-local-8099-2020-06-16-14_54_26

### CODE REVIEW

File: static/scss/cms/catalog-section.scss [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      margin: 0 0 40px;
+
+      @include media-breakpoint-down(sm) {
+        font-size: 15px;
+        margin-bottom: 20px;
+      }
+    }
+
+    .deadline-text {
+      color: #434343;
```
Comment (asadiqbal08) [inline static/scss/cms/catalog-section.scss:56]: Please define in `_variables.scss`

File: cms/templates/partials/home-page-catalog.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{% load wagtailimages_tags image_version_url wagtailcore_tags %}
+<div class="catalog-section">
+    <div class="container">
+        <div class="row">
+            {% for page_block in page.contents %}
+                {% with block=page_block.value %}
+                    <div class="col-sm-6 col-lg-4 p-0 d-flex flex-column">
+
+                        <img src="{% image_version_url block.page.thumbnail_image "fill-475x300" %}" alt="{{ block.page.title }}">
```
Comment (asadiqbal08) [inline cms/templates/partials/home-page-catalog.html:9]: `thumbnail_image` is not marked required so better to add condition else can raise exception.

File: static/scss/cms/catalog-section.scss [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    @include media-breakpoint-down(sm) {
+      padding: 0 15px;
+      font-size: 15px;
+    }
+
+    h2 {
+      font-size: 30px;
+      line-height: 30px;
+      margin: 0 0 40px;
+      color: #505050;
```
Comment (asadiqbal08) [inline static/scss/cms/catalog-section.scss:30]: `matterhorn`

File: static/scss/cms/catalog-section.scss [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      }
+    }
+
+    p {
+      font-weight: 300;
+      margin: 0 0 28px;
+    }
+
+    .dates {
+      color: #737373;
```
Comment (asadiqbal08) [inline static/scss/cms/catalog-section.scss:45]: Please define in `_variables.scss`

File: static/scss/cms/catalog-section.scss [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    display: block;
+    width: 100%;
+    height: auto;
+    margin: 0 0 28px;
+  }
+
+  .info {
+    position: relative;
+    padding: 0 15px 0 60px;
+    color: #191919;
```
Comment (asadiqbal08) [inline static/scss/cms/catalog-section.scss:17]: better to define

File: static/scss/cms/catalog-section.scss (hunk 1 above)
Comment (ahmed-belal) [reply static/scss/cms/catalog-section.scss:56]: Updated

File: cms/templates/partials/home-page-catalog.html (hunk 2 above)
Comment (ahmed-belal) [reply cms/templates/partials/home-page-catalog.html:9]: Updated with `mit-dome.png` as default

File: static/scss/cms/catalog-section.scss (hunk 3 above)
Comment (ahmed-belal) [reply static/scss/cms/catalog-section.scss:30]: Updated

File: static/scss/cms/catalog-section.scss (hunk 4 above)
Comment (ahmed-belal) [reply static/scss/cms/catalog-section.scss:45]: Updated

File: static/scss/cms/catalog-section.scss (hunk 5 above)
Comment (ahmed-belal) [reply static/scss/cms/catalog-section.scss:17]: Updated

File: cms/templates/partials/home-page-catalog.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+<div class="catalog-section">
+    <div class="container">
+        <div class="row">
+            {% for page_block in page.contents %}
+                {% with block=page_block.value %}
+                    <div class="col-sm-6 col-lg-4 p-0 d-flex flex-column">
+                        {% if block.page.thumbnail_image %}
+                            <img src="{% image_version_url block.page.thumbnail_image "fill-475x300" %}" alt="{{ block.page.title }}">
+                        {% else %}
+                            <img src="{% static 'images/mit-dome.png' %}" alt="{{ block.page.title }}">
```
Comment (asadiqbal08) [inline cms/templates/partials/home-page-catalog.html:11]: better to get rid of prefix `mit` in bootcamp.

Comment (asadiqbal08) [reply cms/templates/partials/home-page-catalog.html:11]: ok as per offline conversation, ok with mit-dome.

**Pull request merged**